lib/matplotlib/tests/test_legend.py

```diff
@@ -1667,3 +1667,80 @@ def test_boxplot_legend_labels():
     bp4 = axs[3].boxplot(data, label='box A')
     assert bp4['medians'][0].get_label() == 'box A'
     assert all(x.get_label().startswith("_") for x in bp4['medians'][1:])
+
+
+def test_patchcollection_legend():
+    # Test that PatchCollection labels show up in legend (issue #23998)
+    fig, ax = plt.subplots()
+
+    pc = mcollections.PatchCollection(
+        [
+            mpatches.Polygon([[0, 0], [100, 100], [200, 0]]),
+            mpatches.Polygon([[400, 0], [500, 100], [600, 0]]),
+        ],
+        label="patch collection", facecolor='blue', edgecolor='black'
+    )
+    ax.add_collection(pc, autolim=True)
+    ax.autoscale_view()
+
+    leg = ax.legend()
+
+    # Check that the legend contains our label
+    assert len(leg.get_texts()) == 1
+    assert leg.get_texts()[0].get_text() == "patch collection"
+
+    # Check that the legend handle exists and has correct colors
+    assert len(leg.legend_handles) == 1
+    legend_patch = leg.legend_handles[0]
+    assert_allclose(legend_patch.get_facecolor()[:3],
+                    pc.get_facecolor()[0][:3], rtol=1e-5)
+    assert_allclose(legend_patch.get_edgecolor()[:3],
+                    pc.get_edgecolor()[0][:3], rtol=1e-5)
+
+
+def test_patchcollection_legend_properties():
+    # Test that PatchCollection legend preserves visual properties
+    fig, ax = plt.subplots()
+
+    pc = mcollections.PatchCollection(
+        [mpatches.Circle((0, 0), 1), mpatches.Circle((2, 0), 1)],
+        label="styled patches",
+        facecolor='red',
+        edgecolor='blue',
+        linewidths=3,
+        linestyle='--',
+    )
+    ax.add_collection(pc)
+    ax.autoscale_view()
+
+    leg = ax.legend()
+
+    # Verify that visual properties are preserved
+    legend_patch = leg.legend_handles[0]
+    assert_allclose(legend_patch.get_facecolor()[:3],
```

Comment:
why is alpha getting ignored here?

Reply:
It shouldn't be. The alpha channel should also be verified. I'll change it to check all 4 RGBA components instead of just RGB.